tests/unit/helpers/users.test.js

```diff
@@ -30,15 +31,117 @@ describe('authenticate', () => {
   let UserMock;
   let compare;
   let encode;
+  let momentToDate;
   beforeEach(() => {
     UserMock = sinon.mock(User);
     compare = sinon.stub(bcrypt, 'compare');
     encode = sinon.stub(AuthenticationHelper, 'encode');
+    momentToDate = sinon.stub(momentProto, 'toDate');
   });
   afterEach(() => {
     UserMock.restore();
     compare.restore();
     encode.restore();
+    momentToDate.restore();
+  });
+
+  it('should return authentication data', async () => {
+    const payload = { email: 'toto@email.com', password: 'toto', origin: 'webapp' };
+    const user = {
+      _id: new ObjectID(),
+      refreshToken: 'token',
+      local: { password: 'toto' },
+    };
+    UserMock.expects('findOne')
+      .withExactArgs({ 'local.email': payload.email.toLowerCase() })
+      .chain('select')
+      .withExactArgs('local refreshToken')
+      .chain('lean')
+      .once()
+      .returns(user);
+    compare.returns(true);
+    encode.returns('token');
+    UserMock.expects('updateOne').never();
+
+    const result = await UsersHelper.authenticate(payload);
+
+    expect(result).toEqual({ token: 'token', refreshToken: user.refreshToken, user: { _id: user._id.toHexString() } });
+    UserMock.verify();
+    sinon.assert.calledWithExactly(compare, payload.password, 'toto');
+    sinon.assert.calledWithExactly(
+      encode,
+      { _id: user._id.toHexString() },
+      TOKEN_EXPIRE_TIME
+    );
+  });
+
+  it('should create the field firstMobileConnection for user', async () => {
+    const payload = { email: 'toto@email.com', password: 'toto', origin: 'mobile' };
+    const user = {
+      _id: new ObjectID(),
+      refreshToken: 'token',
+      local: { password: 'toto' },
+    };
+    momentToDate.returns('2020-12-08T13:45:25.437Z');
+
+    UserMock.expects('findOne')
+      .withExactArgs({ 'local.email': payload.email.toLowerCase() })
+      .chain('select')
+      .withExactArgs('local refreshToken')
+      .chain('lean')
+      .once()
+      .returns(user);
+    compare.returns(true);
+    encode.returns('token');
+    UserMock.expects('updateOne')
+      .withExactArgs(
+        { _id: user._id, firstMobileConnection: { $exists: false } },
+        { $set: { firstMobileConnection: '2020-12-08T13:45:25.437Z' } }
+      );
+
+    const result = await UsersHelper.authenticate(payload);
+
+    expect(result).toEqual({ token: 'token', refreshToken: user.refreshToken, user: { _id: user._id.toHexString() } });
+    UserMock.verify();
+    sinon.assert.calledWithExactly(compare, payload.password, 'toto');
+    sinon.assert.calledWithExactly(
+      encode,
+      { _id: user._id.toHexString() },
+      TOKEN_EXPIRE_TIME
+    );
+  });
+
+  it('should create the field firstMobileConnection for user', async () => {
```

Comment:
titre a changer

Reply:
et je ne suis pas sur de comprendre ce test... il pourrait aller dans le 1er cas non ? 
pour moi tu as 6 cas : 
- should authenticate user and set firstMobileConnection
- should authenticate user but not set firstMobileConnection
- les 4 cas du throw error

Reply:
Il ya deux cas enfait dans le not setfirstMobileConnection: 
- ne pas le set si on se connecte de la webapp
- ne pas le set si il existe deja

Reply:
ok du coup est ce que tu pourrais renommer les it stp car tel que c'est je ne trouve pas ca clair ? par exemple
- should authenticate user and set firstMobileConnection
- should authenticate user but not set firstMobileConnection (authentication from webapp)
- should authenticate user but not set firstMobileConnection (firstMobileConnection already set)